### PR TITLE
update footnotes and update asciidoctor version

### PR DESCRIPTION
**Pull request by kwsutter**
Signed-off-by: Kevin Sutter <kwsutter@gmail.com>

This PR should resolve Issue #77.  Had to update the version of the asciidoctor maven plugin in order to get consistent results for the pdf and html generation.  All of the footnotes need to be defined inline.  Not the prettiest asciidoc, but the end result is very good.

### CODE REVIEW

File: specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc [hunk 1]
```diff
@@ -304,7 +306,10 @@ implementation includes support for a technology marked as Optional,
 that technology must be supported in the containers specified above. If
 a product implementation does not support a technology marked as
 Optional, it must not include the APIs for that
-technology.<<a3665, 18>>
+technology.footnote:[Note that a component specification is permitted to specify
+an exception to this in order to accommodate interface type dependencies—for example,
+the Jakarta™ Enterprise Beans SessionContext dependency on the
+javax.xml.rpc.handler.MessageContext type.]
```
Comment (NottyCode) [inline specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc]: this is a really minor nit, but I'd probably mark javax.xml.rpc.handler.MessageContext up using __ as types are elsewhere.

File: specification/src/main/asciidoc/platform/RelatedDocuments.adoc [hunk 2]
```diff
@@ -86,154 +86,45 @@ _Jakarta™ Concurrency Specification, Version 1.0_. Available at: _https://jaka
 _Jakarta™ Batch Specification, Version 1.0_. Available at: _https://jakarta.ee/specifications/batch/1.0_
 
 Extension Mechanism Architecture, Available at
-_http://docs.oracle.com/javase/8/docs/technotes/guides/extensions/index.html_
-.
+_http://docs.oracle.com/javase/8/docs/technotes/guides/extensions/index.html_.
```
Comment (NottyCode) [inline specification/src/main/asciidoc/platform/RelatedDocuments.adoc]: minor nit but we should probably update these to https. I'm happy to create a separate PR for that if you agree.

File: specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc [hunk 3]
```diff
@@ -417,8 +419,11 @@ resource injection, as well as PostConstruct and PreDestroy callbacks.
 In addition, if CDI is enabled—which it is by default—these classes also
 support CDI injection, as described in
 <<a2112, Support for Dependency
```
Comment (NottyCode) [inline specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc]: elsewhere you've been removing the <<a2112, syntax, but it doesn't appear to be fully removed here. Have I misunderstood the change, or is this one needing a little tidying up?

File: specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc [hunk 4]
```diff
@@ -417,8 +419,11 @@ resource injection, as well as PostConstruct and PreDestroy callbacks.
...
-Injection>>”, and the use of interceptors<<a3652, 5>>. The
-component classes listed with support level “Limited” only support Jakarta
+Injection>>”, and the use of interceptors.footnote:[Note that the use of
+interceptors defined by means of the Interceptors annotation is supported in the
+absence of Jakarta™ Contexts and Dependency Injection for Jakarta™ Enterprise
```
Comment (NottyCode) [inline specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc]: I thought we could use CDI in the specs still?

File: specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc (hunk 1 above)
Comment (kwsutter) [reply specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc]: Nice catch.  I just did a cut-n-paste from the existing footnotes.  Thanks.

File: specification/src/main/asciidoc/platform/RelatedDocuments.adoc (hunk 2 above)
Comment (kwsutter) [reply specification/src/main/asciidoc/platform/RelatedDocuments.adoc]: I'll just do it.  I'm already making other changes.

File: specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc (hunk 3 above)
Comment (kwsutter) [reply specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc]: Nope, this is the normal syntax for references.  Look for [[a2112]] for the link.

The other ones that I changed were place holders that I did previously as a reminder for the footnote work that needed to be done.

File: specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc (hunk 4 above)
Comment (kwsutter) [reply specification/src/main/asciidoc/platform/ResourcesNamingInjection.adoc]: Agree, will update.

**Comment by kwsutter** [comment]
For some reason, @NottyCode doesn't have write access so his approval is not sufficient for me to merge.  I've sent a note to the @eclipsewebmaster asking for this to get fixed since he's listed as a Committer on the Platform project.  In the mean time, can I get a quick Approval so that we can finish up this Platform project?  Thanks!

**Comment by pzygielo** [comment]
Could you also update README and bump required maven, please? 
https://github.com/eclipse-ee4j/jakartaee-platform/blob/6821fc84948f08075092f901101d153039b5551f/specification/README.md#L11-L14

Due to:
> [ERROR] Failed to execute goal org.asciidoctor:asciidoctor-maven-plugin:1.5.7.1:process-asciidoc (platform-asciidoc-to-html) on project platform-spec: The plugin org.asciidoctor:asciidoctor-maven-plugin:1.5.7.1 requires Maven version 3.0.5

**Comment by kwsutter** [comment]
> Could you also update README and bump required maven, please?
> 
> https://github.com/eclipse-ee4j/jakartaee-platform/blob/6821fc84948f08075092f901101d153039b5551f/specification/README.md#L11-L14
> 
> Due to:
> 
> > [ERROR] Failed to execute goal org.asciidoctor:asciidoctor-maven-plugin:1.5.7.1:process-asciidoc (platform-asciidoc-to-html) on project platform-spec: The plugin org.asciidoctor:asciidoctor-maven-plugin:1.5.7.1 requires Maven version 3.0.5

Thanks for the info.  Since this PR is reviewed and ready to merge, I don't want to modify the README via this PR.  We'll do it on the next one.

**Pull request merged**